qemu.go

```diff
@@ -662,6 +675,64 @@ func (q *qemu) togglePausePod(pause bool) error {
 	return nil
 }
 
+func (q *qemu) qmpSetup() (*ciaoQemu.QMP, error) {
+	cfg := ciaoQemu.QMPConfig{Logger: qmpLogger{}}
+
+	// Auto-closed by QMPStart().
+	disconnectCh := make(chan struct{})
+
+	qmp, _, err := ciaoQemu.QMPStart(q.qmpControlCh.ctx, q.qmpControlCh.path, cfg, disconnectCh)
+	if err != nil {
+		virtLog.Errorf("Failed to connect to QEMU instance %v", err)
+		return nil, err
+	}
+
+	err = qmp.ExecuteQMPCapabilities(q.qmpMonitorCh.ctx)
+	if err != nil {
+		virtLog.Errorf("Failed to negotiate capabilities with QEMU %v", err)
+		return nil, err
+	}
+
+	return qmp, nil
+}
+
+func (q *qemu) hotplugBlockDevice(drive Drive) error {
+	defer func(qemu *qemu) {
+		if q.qmpMonitorCh.qmp != nil {
+			q.qmpMonitorCh.qmp.Shutdown()
+		}
+	}(q)
+
+	qmp, err := q.qmpSetup()
+	if err != nil {
+		return err
+	}
+
+	q.qmpMonitorCh.qmp = qmp
+
+	if err := q.qmpMonitorCh.qmp.ExecuteBlockdevAdd(q.qmpMonitorCh.ctx, drive.File, drive.ID); err != nil {
+		return err
+	}
+
+	devID := "virtio-" + drive.ID
+	driver := "virtio-blk-pci"
+	if err := q.qmpMonitorCh.qmp.ExecuteDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, ""); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (q *qemu) hotplugDevice(devInfo interface{}, devType deviceType) error {
+	switch devType {
+	case blockDev:
+		drive := devInfo.(Drive)
```

Comment:
This needs to use the 2 arg type assertion syntax to avoid a potential runtime panic.

Reply:
This is a programming error rather than a runtime error. Better have the code panic with the type assertion failure which will be quite evident from the backtrace.